From 75e87e26633fbca7b5bc5aaf9a01e279b39a3213 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Wed, 3 Jul 2024 12:29:06 +0200
Subject: [PATCH 0998/1695] drivers: stm32_risaf: cleanup DT related
 definitions

Gather all device tree definitions in bindings include file.
Move related masks from driver to bindings.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 77e6c85f612778a4a2455848f56d1a9bf1f111e2 drivers: stm32_risaf: cleanup DT related definitions]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I148124382200b8930bd562c087e818dc9cf4710e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391108
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/firewall/stm32_risaf.c            | 9 ---------
 core/include/dt-bindings/soc/stm32mp25-risaf.h | 8 ++++++++
 2 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/core/drivers/firewall/stm32_risaf.c b/core/drivers/firewall/stm32_risaf.c
index 775d6afd5..580898f00 100644
--- a/core/drivers/firewall/stm32_risaf.c
+++ b/core/drivers/firewall/stm32_risaf.c
@@ -112,15 +112,6 @@
 					 (BIT((cid)) << \
 					  _RISAF_REG_CIDCFGR_WRENC_SHIFT))
 
-/* Device Tree related definitions */
-#define DT_RISAF_REG_ID_MASK		U(0xF)
-#define DT_RISAF_EN_MASK		BIT(DT_RISAF_EN_SHIFT)
-#define DT_RISAF_SEC_MASK		BIT(DT_RISAF_SEC_SHIFT)
-#define DT_RISAF_ENC_MASK		GENMASK_32(7, 6)
-#define DT_RISAF_PRIV_MASK		GENMASK_32(15, 8)
-#define DT_RISAF_READ_MASK		GENMASK_32(23, 16)
-#define DT_RISAF_WRITE_MASK		GENMASK_32(31, 24)
-
 #define _RISAF_GET_REGION_ID(cfg)	((cfg) & DT_RISAF_REG_ID_MASK)
 #if defined(CFG_STM32MP21)
 #define _RISAF_GET_REGION_CFG(cfg) \
diff --git a/core/include/dt-bindings/soc/stm32mp25-risaf.h b/core/include/dt-bindings/soc/stm32mp25-risaf.h
index ac9d53420..0a634c6a1 100644
--- a/core/include/dt-bindings/soc/stm32mp25-risaf.h
+++ b/core/include/dt-bindings/soc/stm32mp25-risaf.h
@@ -27,6 +27,14 @@
 #define DT_RISAF_READ_SHIFT	16
 #define DT_RISAF_WRITE_SHIFT	24
 
+#define DT_RISAF_REG_ID_MASK		U(0xF)
+#define DT_RISAF_EN_MASK		BIT(DT_RISAF_EN_SHIFT)
+#define DT_RISAF_SEC_MASK		BIT(DT_RISAF_SEC_SHIFT)
+#define DT_RISAF_ENC_MASK		GENMASK_32(7, 6)
+#define DT_RISAF_PRIV_MASK		GENMASK_32(15, 8)
+#define DT_RISAF_READ_MASK		GENMASK_32(23, 16)
+#define DT_RISAF_WRITE_MASK		GENMASK_32(31, 24)
+
 #define RISAFPROT(risaf_region, cid_read_list, cid_write_list, cid_priv_list, sec, enc, enabled) \
 	(((cid_write_list) << DT_RISAF_WRITE_SHIFT) | \
 	 ((cid_read_list) << DT_RISAF_READ_SHIFT) | \
-- 
2.47.2

